runtime/compiler/env/VMJ9.cpp

```diff
@@ -2309,18 +2309,40 @@ TR_MarkHotField::markHotField(J9Class * clazz, bool rootClass)
       J9ROMClass* romClass = TR::Compiler->cls.romClassOf((TR_OpaqueClassBlock*)clazz);
       J9UTF8* name = J9ROMCLASS_CLASSNAME(romClass);
       printf("%*smarked field as hot in class %.*s\n", depth(), " ", J9UTF8_LENGTH(name), J9UTF8_DATA(name));
-      }
-
-   *(UDATA *)((char *)clazz + offsetOfHotFields()) = noncoldWord | _bitValue;
-
-   return true;
+      }  
+
+      if (jitConfig->javaVM->memoryManagerFunctions->j9gc_dynamic_breadth_first_scan_ordering_enabled(jitConfig->javaVM)) {
```

Comment:
I would give a more generic name to this MM helper. For example, j9gc_hot_reference_field_required. In future, we may use it for some other reasons (like Depth-First scan). Accordingly, the comment below should be more generic or even removed.